### DISCUSSION TITLE
More User Documentation?

### DISCUSSION THREAD

**Discussion by juliomac** [infoderm/patients#550] [Q&A] [answered]
Very nice! I would like to contrite an also use it...

I Would like understand:
1) How to add Doctors ?
2) Is it possible to have Patients scheduling consults outside the app (from an external website / route)?
3) i18n contribution ?
4) User Documentation? Where is it being used?

**Comment by make-github-pseudonymous-again** [comment, marked as answer]
Hi Júlio!

In general, I am interested in collaboration/contribution.
Currently there is no user documentation. But I can answer your questions.

> 1. How to add Doctors ?
> 2. Is it possible to have Patients scheduling consults outside the app (from an external website / route)?
> 3. i18n contribution ?
> 4. User Documentation? Where is it being used?

1. Do you mean app users or patient doctors? New users registration is colocated with the login functionality in a pop-up. Patient doctors are created on the fly when adding not-yet-known patient doctors to a patient's personal information. App users and patient doctors are not the same thing. Patient doctors is really only patient information for the moment. App users and patient doctors cannot be "linked".
2. It is currently not possible for anybody other than the user to schedule appointments. Users are siloed, so it's not even currently possible to have other healthcare professionals to schedule appointments for the user.
3. The only i18n effort made so far is date localization. Also currently the time availability feature only works when both the server and the client share the same timezone (see #512, 97dfc23b8de9453f2215ec7ac358733d05b1b609 and c305f6bceb2fa300f9aeda935e3e4e0551495677). The main issue for UI i18n is #300. I am still not decided on which tool to use.
4. I currently only have a single user who has been using the app for four years now. There is no written user documentation.

I am interested in contributions on all four topics, although my first concern is i18n as I would like this app to have perfect support for English, French, and Dutch (targeting belgian users). I also have to warn you that certain functionalities right now only cater for belgian specificities (patient ID cards, lab reports formats, etc.), although they could probably be made more generic without too much effort.

**Reply by juliomac** [reply]
Thanks for you answer! 

Regarding i18n, I have tried and used many different i18n packages overtime. They are nice, but the problem with all of them is that they get old and loose support with time.

Now that HTML 5 and javascript are filled with many I18n tools, I would go with no package at all. If you check this article [https://phrase.com/blog/posts/step-step-guide-javascript-localization/](url), there is a very good step by step, very advanced example, on how to go with HTML 5 and plain javascript. The example is very similar in terms of easy to use to most packages anyway.

Cooperation:
I understand that you had a set of goals when you started developing this app. I guess that if somehow ours and your goals get some sort of alignment, maybe we could cooperate.

My main goal is to have a simple tool for a small clinic that: 1) Facilitate the scheduling and management of appointments for Doctors. That would include a Web interface where Patients could easily make an appointment, in sync with each Doctors agenda, following Doctors restrictions in terms of time slots and availability. (configurable by Doctors / assistants) 2) Manage patient medical history and Patient information. 

I guess that old require Doctors to be an entity/user and not just a information file. Doctors and assistants would have a role in managing the Agenda, being able add, delete and edit. Patients could be able only to see blocked / available slots and request / select a date/time-slot, at their convenience, for a choosen Doctor. That would be fed directly into that Doctor's agenda.

Would that make sense to you as well?

**Reply by make-github-pseudonymous-again** [reply]
> My main goal is to have a simple tool for a small clinic that: 1) Facilitate the scheduling and management of appointments for Doctors. That would include a Web interface where Patients could easily make an appointment, in sync with each Doctors agenda, following Doctors restrictions in terms of time slots and availability. (configurable by Doctors / assistants) 2) Manage patient medical history and Patient information.
> I guess that old require Doctors to be an entity/user and not just a information file. Doctors and assistants would have a role in managing the Agenda, being able add, delete and edit. Patients could be able only to see blocked / available slots and request / select a date/time-slot, at their convenience, for a chosen Doctor. That would be fed directly into that Doctor's agenda.

Yep that's a feature that could be implemented and that would be useful to certain users. Although my current only user would like to disable the functionality.

I think the difficulty with implementing this would lie in defining read/write permissions for users (see for instance #276).

> Regarding i18n, I have tried and used many different i18n packages overtime. They are nice, but the problem with all of them is that they get old and loose support with time.
> Now that HTML 5 and javascript are filled with many I18n tools, I would go with no package at all. If you check this article [https://phrase.com/blog/posts/step-step-guide-javascript-localization/](https://github.com/infoderm/patients/discussions/url), there is a very good step by step, very advanced example, on how to go with HTML 5 and plain javascript. The example is very similar in terms of easy to use to most packages anyway.

Going to read about this. Thanks!

**Reply by juliomac** [reply]
Regarding your only user unwillingness of patients use of self scheduling, the solution would be simply not publishing it at its website. For permissions, take a look at the popular package 'alanning:roles' now maintained as a Community package at [https://github.com/Meteor-Community-Packages/meteor-roles](url)
